pkg/ctl/gitops/apply.go

```diff
@@ -0,0 +1,178 @@
+package gitops
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"k8s.io/client-go/tools/clientcmd"
+
+	kubeclient "k8s.io/client-go/kubernetes"
+
+	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
+	"github.com/weaveworks/eksctl/pkg/ctl/cmdutils"
+	"github.com/weaveworks/eksctl/pkg/git"
+	"github.com/weaveworks/eksctl/pkg/gitops"
+	"github.com/weaveworks/eksctl/pkg/gitops/fileprocessor"
+	"github.com/weaveworks/eksctl/pkg/gitops/flux"
+)
+
+const (
+	defaultGitTimeout = 20 * time.Second
+)
+
+type options struct {
+	git               git.Options
+	quickstartNameArg string
+	outputPath        string
+}
+
+func applyGitops(cmd *cmdutils.Cmd) {
+	cfg := api.NewClusterConfig()
+	cmd.ClusterConfig = cfg
+
+	cmd.SetDescription("apply", "Setting up GitOps and apply a Quick Start profile", "")
+
+	var opts options
+
+	cmd.SetRunFuncWithNameArg(func() error {
+		return doApplyGitops(cmd, opts)
+	})
+
+	cmd.FlagSetGroup.InFlagSet("General", func(fs *pflag.FlagSet) {
+		fs.StringVarP(&opts.quickstartNameArg, "quickstart-profile", "", "", "name or URL of the Quick Start profile. For example, app-dev.")
+		fs.StringVarP(&opts.git.URL, "git-url", "", "", "URL for the git repository that will contain the cluster components")
+		fs.StringVarP(&opts.git.Branch, "git-branch", "", "master", "Git branch")
+		fs.StringVarP(&opts.outputPath, "output-path", "", "./", "Path to directory where the GitOps repo will be cloned")
+		fs.StringVar(&opts.git.User, "git-user", "Flux", "Username to use as Git committer")
+		fs.StringVar(&opts.git.Email, "git-email", "", "Email to use as Git committer")
+		fs.StringVar(&cfg.Metadata.Name, "cluster", "", "name of the EKS cluster to add the nodegroup to")
+		_ = cobra.MarkFlagRequired(fs, "quickstart-profile")
+		_ = cobra.MarkFlagRequired(fs, "git-url")
```

Comment:
Hah, I didn't know cobra allowed for this, we should use it in `eksctl install flux`